model/span/event.go

```diff
@@ -273,23 +273,23 @@ func (d *DestinationService) fields() common.MapStr {
 	return fields
 }
 
-func DecodeRUMV3Event(input interface{}, cfg m.Config, err error) (transform.Transformable, error) {
-	return DecodeEvent(input, cfg, err)
+func DecodeRUMV3Event(input m.Input, err error) (transform.Transformable, error) {
+	return DecodeEvent(input, err)
 }
 
 // DecodeEvent decodes a span event.
-func DecodeEvent(input interface{}, cfg m.Config, err error) (transform.Transformable, error) {
+func DecodeEvent(input m.Input, err error) (transform.Transformable, error) {
 	if err != nil {
 		return nil, err
 	}
-	if input == nil {
+	if input.Raw == nil {
 		return nil, errMissingInput
 	}
-	raw, ok := input.(map[string]interface{})
+	raw, ok := input.Raw.(map[string]interface{})
```

Comment:
Maybe move all this logic duplicated on each event to somewhere around https://github.com/elastic/apm-server/pull/3570/files#diff-01c58acd54a543ec500d1ebf9e7bbe94R191 (ie. let `input.Raw` be a non nil `map[string]interface{}`

Reply:
Sounds good, I will do that in a follow-up.